ereolen_support/ereolen_support.helper.inc

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * @file
+ * Contains helper functions for the support form.
+ */
+
+/**
+ * Search for a drupal user email in Jira.
+ *
+ * @param \stdClass $user
+ *   A drupal user.
+ *
+ * @return bool|mixed
+ *   A jira user or FALSE if no user was found.
+ *
+ * @throws \Exception
+ */
+function _ereolen_support_search_jira_user($user, $form_state) {
+  $form_email = $form_state['values']['ereolen_support_email'];
+  $requestParts = _ereolen_support_get_request_parts('GET');
+  $result = drupal_http_request($requestParts['jira_path'] . '/user/search?includeInactive=true&username=' . $form_email, $requestParts['options']);
+  $resultArray = json_decode($result->data);
+  _ereolen_support_log_error($result);
+
+  // Searches for matches in full name, username and email, so we may have
+  // multiple results.
+  if(!empty($resultArray)) {
+    foreach ($resultArray as $jiraUser) {
+      if($jiraUser->name == $form_email) {
+        // If user is deactivated in Jira we need to activate it.
+        if ($jiraUser->active == false) {
```

Comment:
Is there an way to validate which type of user we are re-activating here... so it's not an user with an false mail address that tries to trig us into opening an account that have to much access.

Reply:
okay, a comment about these workflows would be nice... I did not know that they exists. So that means that if my user was deactivated and re-activated through this call... I would not have the same permissions and be able to log-into jira?

Reply:
If a user with a license is deactivated they are stripped from all roles to prevent them from filling up a license. If the user is then reactivated, they still do not have any roles. Also. Users adding issues through this module are by default not granted any permissions in the first place.

In case of your user:
Jesper is deactivated and stripped of all roles in the Jira system
When Jesper is reactivated through this call Jesper has no permissions, but as he is now recognized by Jira as a user he is able to receive mails.